src/dune_rules/ml_sources.ml

```diff
@@ -505,6 +528,12 @@ let modules_of_stanzas =
     in
     `Executables { Per_stanza.stanza = exes; sources; modules; obj_dir; dir }
   in
+  let make_tests ~dir ~expander ~modules ~project tests =
+    let+ result = make_executables ~dir ~expander ~modules ~project tests.Tests.exes in
+    match result with
+    | `Executables group_part -> `Tests { group_part with stanza = tests }
+    | _ -> assert false
```

Comment:
I don't get why this is an `assert false`, what other constructors can this return?